### PR TITLE
Cross platform e2e

### PR DESCRIPTION
**Pull request by wcs1only**
# Description

The dapr control plane and e2e tests can now be run on windows containers. Simply set TARGET_OS=windows on a docker enabled windows machine and follow the instructions https://github.com/dapr/dapr/blob/master/tests/docs/running-e2e-test.md. Your Kubernetes cluster must have a windows node pool, of course.

I have run the tests manually, and they pass. A separate PR will follow shortly to integrate this into our CI/CD pipeline, once I have created the appropriate node pools in our test infrastructure.
```
=== RUN   TestStateApp                                                                                                                                                                                                                  [1240/10778]
2020/08/14 02:57:50 Waiting until service ingress is ready for stateapp...                                                                                                                                                                          
2020/08/14 02:57:57 Service ingress for stateapp is ready...                                                                                                                                                                                        
=== RUN   TestStateApp/Test_get_getbulk_save_delete_with_empty_request_response_for_single_app_and_single_hop                                                                                                                                       
=== RUN   TestStateApp/Test_save_get_and_delete_a_single_item_for_single_app_and_single_hop                               
=== RUN   TestStateApp/Test_save_get_and_delete_on_multiple_items_for_single_app_and_single_hop                                                                                                                                                     
--- PASS: TestStateApp (44.47s)                                                                                                                                                                                                                     
    --- PASS: TestStateApp/Test_get_getbulk_save_delete_with_empty_request_response_for_single_app_and_single_hop (0.34s)                                                                                                                           
    --- PASS: TestStateApp/Test_save_get_and_delete_a_single_item_for_single_app_and_single_hop (0.03s)                                                                                                                                             
    --- PASS: TestStateApp/Test_save_get_and_delete_on_multiple_items_for_single_app_and_single_hop (0.06s)               
PASS                                                                                                                                                                                                                                                
2020/08/14 02:58:35 Running teardown...                                                                                                                                                                                                             
ok      github.com/dapr/dapr/tests/e2e/stateapp 116.446s                                                                                                                                                                                            
?       github.com/dapr/dapr/tests/e2e/utils    [no test files]  
```

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1747 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by wcs1only** [comment]
Seems to be a transient failure e2e tests? Not in a test that I made any changes to, and also happening on other branches.

**Comment by yaron2** [comment]
> Seems to be a transient failure e2e tests? Not in a test that I made any changes to, and also happening on other branches.

Yes, the new transaction e2e tests from [this](https://github.com/dapr/dapr/pull/1892) PR are failing.

**Comment by yaron2** [comment]
> > Seems to be a transient failure e2e tests? Not in a test that I made any changes to, and also happening on other branches.
> 
> Yes, the new transaction e2e tests from [this](https://github.com/dapr/dapr/pull/1892) PR are failing.

I [opened a PR](https://github.com/dapr/dapr/pull/1908) that will make the test fail instead of panic. This doesn't explain why the transactional e2e test is not passing.

**Comment by youngbupark** [comment]
> > > Seems to be a transient failure e2e tests? Not in a test that I made any changes to, and also happening on other branches.
> > 
> > 
> > Yes, the new transaction e2e tests from [this](https://github.com/dapr/dapr/pull/1892) PR are failing.
> 
> I [opened a PR](https://github.com/dapr/dapr/pull/1908) that will make the test fail instead of panic. This doesn't explain why the transactional e2e test is not passing.

@yaron2 @wcs1only master branch is reproducing so we can merge this one. then I could work separately.

File: tests/platforms/kubernetes/kubeobj.go [hunk 1]
```diff
@@ -32,11 +32,20 @@ const (
...
+
+	// Environment variable for setting Kubernetes node affinity ARCH
+	TargetArchEnvVar = "TARGET_ARCH"
 )
 
 // DaprTestNamespace is the default Kubernetes namespace for e2e tests
 var DaprTestNamespace = "dapr-tests"
 
+var TargetOs = "linux"
+var TargetArch = "amd64"
```
Comment (youngbupark) [inline tests/platforms/kubernetes/kubeobj.go:47]: nit:
we rewrite this way:
``` 
var (
   DaprTestNamespace = "dapr-tests"
   TargetOs = "linux"
   TargetArch = "amd64"
)
```

Comment (wcs1only) [reply tests/platforms/kubernetes/kubeobj.go:47]: Oh sure. Forgive me, I'm new to go :).

**Pull request merged**